system/doc/design_principles/sup_princ.md

```diff
@@ -489,14 +491,15 @@ child_spec() = #{id => child_id(),             % mandatory
   `worker`, the default value `5000` will be used; if the child is of type
   `supervisor`, the default value `infinity` will be used.
 
-- `type` specifies if the child process is a supervisor or a worker.
+- `type` specifies whether the child process is a supervisor or a worker.
 
   The `type` key is optional. If it is not given, the default value `worker`
   will be used.
 
-- `modules` are to be a list with one element `[Module]`, where `Module` is the
-  name of the callback module, if the child process is a supervisor, gen_server,
-  gen_statem. If the child process is a gen_event, the value shall be `dynamic`.
+- `modules` are to be a list with a single element `[Module]`, where
+  `Module` is the name of the callback module, if the child process is
+  a supervisor, a gen_server, or a gen_statem. If the child process is a
+  gen_event, the value has to be `dynamic`.
```

Comment:
```suggestion
- `modules` is to be a list with a single element `[Module]`, where
  `Module` is the name of the callback module, if the child process is
  a supervisor, a gen_server, or a gen_statem. If the child process is a
  gen_event, the value has to be the atom `dynamic`.
```
("a list", ie singular)

However, the use of "is to be" sounds kinda strange (to me at least).

Also, the first sentence is quite comma-heavy with the sub-sentences and enumerations. Maybe this should be rephrased completely?

Reply:
Yes, you are right. I have rephrased it.